doc/advopt.txt

```diff
@@ -45,14 +45,17 @@ Advanced options:
   --spellSuggest|:num       show at most `num >= 0` spelling suggestions on typos.
                             if `num` is not specified (or `auto`), return
                             an implementation defined set of suggestions.
-  -w:on|off|list, --warnings:on|off|list
-                            turn all warnings on|off or list all available
-  --warning[X]:on|off       turn specific warning X on|off
-  --hints:on|off|list       turn all hints on|off or list all available
-  --hint[X]:on|off          turn specific hint X on|off
-  --warningAsError[X]:on|off
+  --hints:on|off|all|none|list
+                            `on|off` enables or disables currently selected hints and acts
+                            as a reversible gate, whereas `all|none` selects or deselects
+                            all hints. `list` reports which hints are selected.
+  --hint:X:on|off           turn specific hint X on|off. `hint:X` means `hint:X:on`.
+  -w:on|off|all|none|list, --warnings:...
+                            same as `--hints` but for warnings.
+  --warning:X:on|off        turn specific warning X on|off
+  --warningAsError:X:on|off
                             turn specific warning X into an error on|off
-  --hintAsError[X]:on|off   turn specific hint X into an error on|off
+  --hintAsError:X:on|off    turn specific hint X into an error on|off
   --styleCheck:off|hint|error
```

Comment:
i also changed `--warning[X]:on:off` to `--warning:X:on:off` in help (both are already supported before this PR, the latter is newer and preferred because it doesn't require shell quoting)